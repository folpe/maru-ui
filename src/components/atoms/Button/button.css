/* reset */
button,
input[type='submit'],
input[type='reset'] {
  background: none;
  color: inherit;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  outline: inherit;
}

button,
input[type='submit'],
input[type='reset'] {
  font-size: 12px;
  padding: 8px 16px;
  border-radius: 10px;
}

/* CONTAINED */
button.contained.primary {
  color: white;
  background: red;
}

button.contained.secondary {
  color: white;
  background: blue;
}

/* OUTLINED */
button.outlined.primary {
  color: red;
  border: 1px solid red;
}

button.outlined.secondary {
  color: blue;
  border: 1px solid blue;
}

/* TEXT */
button.primary {
  color: red;
}

button.secondary {
  color: blue;
}

/* SIZE */
button.small {
  font-size: 1em;
}

button.medium {
  font-size: 1.4em;
}

button.large {
  font-size: 1.8em;
}

/* FULL */
button.full {
  width: 100%;
}
