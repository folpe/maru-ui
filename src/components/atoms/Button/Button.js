import React from 'react'
import { bool, node, oneOf } from 'prop-types'
import classnames from 'classnames'

import './button.css'

const Button = (
  {
    children,
    variant = 'text',
    color = 'primary',
    size = 'small',
    full = false,
  },
  props
) => {
  return (
    <button {...props} className={classnames(variant, color, size, { full })}>
      {children}
    </button>
  )
}

Button.propTypes = {
  children: node.isRequired,
  variant: oneOf(['contained', 'outlined', 'text']),
  color: oneOf(['primary', 'secondary']),
  size: oneOf(['small', 'medium', 'large']),
  full: bool,
}

export default Button
