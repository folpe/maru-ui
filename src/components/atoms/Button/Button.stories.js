import React from 'react'

import Button from './Button'

export default {
  title: 'Atoms/Button',
  component: Button,
}

const Template = (args) => <Button {...args}>Valider</Button>

export const Variant = Template.bind({})
Variant.args = {
  variant: 'text',
  children: 'Toto',
}

export const Default = () => <Button>Coucou</Button>

export const Contained = () => <Button variant='contained'>Hello</Button>
export const Outlined = () => <Button variant='outlined'>Hello</Button>
export const Secondary = () => (
  <Button color='secondary' size='large'>
    Coucou
  </Button>
)
export const Full = () => (
  <Button variant='contained' color='secondary' full>
    Hello
  </Button>
)
