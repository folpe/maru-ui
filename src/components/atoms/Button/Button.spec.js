import React from 'react'
import {render} from '@testing-library/react'

import Button from './Button'

describe('<Button />', () => {

  it('should otherwise match snapshot', ()=> {
    const {container} = render(<Button>Valider</Button>)

    expect(container).toMatchSnapshot()
  })
})
