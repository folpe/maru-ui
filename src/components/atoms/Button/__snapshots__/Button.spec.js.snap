// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should otherwise match snapshot 1`] = `
<div>
  <button
    class="text primary small"
  >
    Valider
  </button>
</div>
`;
